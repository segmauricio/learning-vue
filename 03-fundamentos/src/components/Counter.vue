<template>
    <h2>{{customTitle}}</h2>
    <!--<p>{{ counter }} <sup>2</sup> = {{ counter * counter }}</p>-->
    <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
    <div>
        <button v-on:click="increase">+</button>
        <button @click="decrease">-</button>
    </div>
</template>

<script>
    export default {
        //props: ['title', 'start'],
        props: {
            title: String,
            start: {
                type: Number,
                required: true,
                default: 100,
                validator(value){
                    return value >= 0
                }
            }
        },
        data(){
            return{
                counter: this.start
            }
        },
        methods: {
            getSquareValue(){
                console.log('getSquareValue');
                return this.counter * this.counter
            },
            increase(){
                return this.counter++
            },
            decrease(){
                return this.counter--
            }
        },
        computed: {
            squareCounter(){
                return this.counter * this.counter
            },
            customTitle(){
                return this.title ? this.title : 'Counter'
            }
        }
    };
</script>

<style>
    button{
        background-color: #2fd780;
        border-radius: 3px;
        border: 1px solid whitesmoke;
        color: white;
        cursor: pointer;
        margin: 0 5px;
        padding: 5px 15px;
        transition: 0.3s ease-in-out;
    }
    button:hover{
        background-color: #029047;
        transition: 0.3s ease-in-out;
    }
</style>
