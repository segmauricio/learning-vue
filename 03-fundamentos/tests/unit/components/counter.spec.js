import { shallowMount, mount } from '@vue/test-utils'
import Counter from '@/components/Counter'

describe('Counter Component', () => {
    /*
    test('Debe de hacer match con el snapshot', () => {
        const wrapper = shallowMount(Counter)
        expect( wrapper.html() ).toMatchSnapshot()
     })
     */
    test('h2 debe de tener el valor por defecto "Counter"', () => { 
        const wrapper = shallowMount( Counter )
        expect ( wrapper.find('h2').exists() ).toBeTruthy()
        const h2Value = wrapper.find('h2').text()
        //console.log(h2);
        expect ( h2Value ).toBe('Counter')
        
     })
})