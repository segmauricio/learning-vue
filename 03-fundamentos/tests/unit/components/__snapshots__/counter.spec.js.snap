// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe de hacer match con el snapshot 1`] = `
<div>
  <h2>Counter</h2>
  <!--<p>{{ counter }} <sup>2</sup> = {{ counter * counter }}</p>-->
  <p>100 <sup>2</sup> = 10000</p>
  <div><button>+</button><button>-</button></div>
</div>
`;
